modcc/cudaprinter.cpp

```diff
@@ -694,10 +706,11 @@ void CUDAPrinter::visit(ProcedureExpression *e) {
     }
     else {
         // net_receive() kernel is a special case, not covered by APIMethod visit.
+
+        // Core `net_receive` kernel is called device-side from `kernel::deliver_events`.
         text_.add_gutter() << "template <typename T, typename I>\n";
-        text_.add_line(       "__global__");
-        text_.add_gutter() << "void " << e->name()
-                           << "(" << module_->name() << "_ParamPack<T,I> params_, "
+        text_.add_line(       "__device__");
+        text_.add_gutter() << "void net_receive(const " << module_->name() << "_ParamPack<T,I>& params_, "
```

Comment:
I've added a comment in `src/mechanism.hpp` to the effect that it exposed for testing purposes.